crypto_kdf.js

```diff
@@ -29,7 +29,7 @@ module.exports.crypto_kdf_derive_from_key = function crypto_kdf_derive_from_key
   STORE64_LE(salt, subkey_id)
 
   var outlen = Math.min(subkey.length, module.exports.crypto_kdf_BYTES_MAX)
-  blake2b(outlen, key, salt, ctx_padded, true)
+  blake2b(outlen, key.slice(0, blake2b.KEYBYTES), salt, ctx_padded, true)
```

Comment:
One nitpick; `slice` for a `Buffer` returns a view into the same memory, while `slice` on a `ArrayBuffer` returns a copy. I think `subarray` here would be the most appropriate one, since that always returns a view into the same data, but assumes that a `Uint8Array` is backing the `Buffer` (which it is for newer node)

Reply:
Oh, and can you switch `blake2b.KEYBYTES` to `module.exports.crypto_kdf_KEYBYTES` (defined a bit above).

Reply:
module.exports.crypto_kdf_KEYBYTES is 64 but blake2b.KEYBYTES is 32, so I don't think that will work.

Reply:
Oh,  that's actually a bug here that the constant is 64. It should be 32: https://github.com/jedisct1/libsodium/blob/d64a3c6e2f99f86ec7e927282a06e81dfe552886/src/libsodium/include/sodium/crypto_kdf_blake2b.h#L29